views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -0,0 +1,720 @@
+            });
+            stop.on('click', function() {
+                if (this.is('enabled')) {
+                    if (self.recorder.is('recording')) {
+                        self.stopRecording();
+
+                    } else if (self.player.is('playing')) {
+                        self.stopPlayback();
+                    }
+                }
+            }.bind(stop));
+            stop.on('updatestate', function() {
+                if (self.player.is('playing')
+                    || self.recorder.is('recording')) {
+                    this.enable();
+                } else {
+                    this.disable();
+                }
+            }.bind(stop));
+            this.controls.stop = stop;
+
+
+            // Play button
+            if (this.config.allowPlayback === true) {
+                play = uiElements.controlFactory({
+                    id: 'play',
+                    label: this.getControlIcon('play'),
+                    container: this.$controlsContainer
+                });
+                play.on('click', function() {
+                    if (this.is('enabled')) {
+                        self.playRecording();
+                    }
+                }.bind(play));
+                play.on('updatestate', function() {
+                    if (self.player.is('idle')) {
+                        this.enable();
+                    } else {
+                        this.disable();
+                    }
+                }.bind(play));
+                this.controls.play = play;
+            }
+
+
+            // Reset button
+            if (this.config.maxRecords !== 1) {
+                reset = uiElements.controlFactory({
+                    id: 'reset',
+                    label: this.getControlIcon('reset'),
+                    container: this.$controlsContainer
+                });
+                reset.on('click', function() {
+                    if (this.is('enabled')) {
+                        self.resetRecording();
+                        self.updateResetCount();
+                    }
+                }.bind(reset));
+                reset.on('updatestate', function() {
+                    if (self.config.maxRecords > 1 && self.config.maxRecords === self._recordsAttempts) {
+                        this.disable();
+                    } else if (self.player.is('idle')) {
+                        this.enable();
+                    } else {
+                        this.disable();
+                    }
+                }.bind(reset));
+                this.controls.reset = reset;
+            }
+
+            self.updateControls();
+        },
+
+        /**
+         * Update the state of all the controls
+         */
+        updateControls: function updateControls() {
+            var control;
+            for (control in this.controls) {
+                if (this.controls.hasOwnProperty(control)) {
+                    this.controls[control].updateState();
```

Comment:
Why didn't you use lodash helper to loop ? (`_.forOwn()`, or better `_.invoke()`)